### ISSUE TITLE
ARM: `azure ad app` missing `list|show` and json output is malformed

### ISSUE THREAD

**Issue by colemickens** [Azure/azure-xplat-cli#2216]
I need to create an app from the xplat-cli and then be able to query for the app again to retrieve the ApplicationId that is assigned to it.

This information seems to only be accessible, currently, as you create the app, since `azure ad app list` and `azure ad app show` are not valid commands.

So I decided to try this to create the app and then pull the application id out of the returned json:
`azure ad app create --name "testtest" --password "testtest" --home-page=http://dev/null1 --identifier-uris=http://dev/null1 --json`

but the json that was returned looked like this:

```
"e3efaa65-cb74-4051-b140-7378cac2484a"
"247a0d18-fd3d-4385-b9c1-55fede955d76"
{}
"user_impersonation"
"Allow the application to access testtest on behalf of the signed-in user."
[]
"Access testtest"
[
  {
    "impersonated": "User",
    "impersonator": "Application"
  }
]
{}
"Application"
"02caef30-3b19-4ec2-9912-950c379be885"
"Personal"
"Allow the application to access testtest on your behalf."
"Access testtest"
{}
```

**Comment by colemickens** [comment]
Actually, even without --json, the output looks cut off:

```
$ azure ad app create --name kube-spoke-app --password=6f6e1b60-8b16-4770-bb8c-2bdfd22092ce --home-page=http://kube-spoke/dev/null --identifier-uris=http://kube-spoke/dev/null
info:    Executing command ad app create
info:    Creating application kube-spoke-app
data:    Application Id:          9bdf107b-ef46-469d-bea0-cc6dfab6a273
data:    Application Object Id:   122d1c8b-a863-47cb-b6c8-aa78452ae917
data:    Application Permissions:  
data:                             claimValue:  user_impersonation
data:                             description:  Allow the application to access kube-spoke-app on behalf of the signed-in user.
data:                             directAccessGrantTypes: 
data:                             displayName:  Access kube-spoke-app
data:                             impersonationAccessGrantTypes:  impersonated=User, impersonator=Application
data:                             isDisabled: 
data:                             origin:  Application
data:                             permissionId:  7292c88f-cd99-41ce-9775-0b9608f4f4d9
data:                             resourceScopeType:  Personal
data:                             userConsentDescription:  Allow the application to access kube-spoke-app on your behalf.
data:                             userConsentDisplayName:  Access kube-spoke-app
data:                             lang: 
info:    ad app create command OK'
```

**Comment by yugangw-msft** [comment]
@shuagarw, please take a look. 2 sub items, the "--json option support" and to clarify whether cut-off exists

**Comment by yugangw-msft** [comment]
@vivsriaus , please take a look. There are several json format issue we can fix them at the same time

**Comment by amarzavery** [comment]
All of the above issues have been fixed in #2944 
D:\sdk\azure-xplat-cli>node bin/azure ad app -h
help:    Commands to display Active Directory applications
help:
help:    Creates a new Active Directory application
help:      ad app create [options]
help:
help:    Updates the properties of the created Active Directory application
help:      ad app set [options]
help:
help:    Deletes the Active Directory application
help:      ad app delete [options] <object-id>
help:
help:    Get all Active Directory applications in current subscription's tenant
help:      ad app list [options]
help:
help:    Get Active Directory applications
help:      ad app show [options]
help:
help:    Options:
help:      -h, --help  output usage information
help:
help:    Current Mode: arm (Azure Resource Management)

D:\sdk\azure-xplat-cli>
